File: schemas/sdg_coin_wallet_history.py
from datetime import datetime
from pydantic import BaseModel
from typing import Optional

class SDGCoinWalletHistorySchemaBase(BaseModel):
    history_id: int
    wallet_id: int
    increment: float
    reason: Optional[str]
    is_shown: Optional[bool] = False
    timestamp: datetime

    model_config = {
        "from_attributes": True
    }


class SDGCoinWalletHistorySchemaFull(SDGCoinWalletHistorySchemaBase):
    created_at: datetime
    updated_at: datetime

    model_config = {
        "from_attributes": True
    }

# Not directly derived from models
# Todo: Generate TS type
class NoSDGCoinWalletHistorySchemaBase(BaseModel):
    message: str = "No wallet history found for the user."
    increment: float = 0.0
    reason: Optional[str] = "No updates available."

File: frontend/types/sdgCoinWalletHistory.ts
export interface SDGCoinWalletHistorySchemaBase {
  historyId: number;
  walletId: number;
  increment: number;
  reason?: string | null;
  isShown?: boolean | null;
  timestamp: string; // ISO date string
}

export interface SDGCoinWalletHistorySchemaFull extends SDGCoinWalletHistorySchemaBase {
  createdAt: string;
  updatedAt: string;
}
